### PR TITLE
chore(plugins): no longer include plugin start.php

### PR DESCRIPTION
**Pull request by jdalsem**
BREAKING CHANGE

### CODE REVIEW

File: mod/discussions/classes/Elgg/Discussions/Menus/Site.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	public static function register(\Elgg\Hook $hook) {
+		if (!elgg_get_plugin_setting('enable_global_discussions', 'discussions')) {
+			return;
+		}
+		
+		$return = $hook->getValue();
+		$return[] = \ElggMenuItem::factory([
+			'name' => 'discussions',
+			'text' => elgg_echo('collection:object:discussion'),
+			'href' => elgg_generate_url('collection:object:discussion:all'),
```
Comment (jeabakker) [inline mod/discussions/classes/Elgg/Discussions/Menus/Site.php]: `default:object:discussion` see other remark about site menu item

File: mod/discussions/classes/Elgg/Discussions/Permissions.php [hunk 2]
```diff
@@ -32,4 +32,24 @@ public static function containerLogic(\Elgg\Hook $hook) {
...
+	 * Make sure that discussion comments can not be written to a discussion after it has been closed
+	 *
+	 * @param \Elgg\Hook $hook 'container_logic_check', 'object'
+	 *
+	 * @return void|false
+	 */
+	public static function preventCommentOnClosedDiscussion(\Elgg\Hook $hook) {
+		
+		$discussion = $hook->getEntityParam();
+		if (!$discussion instanceof ElggDiscussion) {
```
Comment (jeabakker) [inline mod/discussions/classes/Elgg/Discussions/Permissions.php]: `\ElggDiscussion`?

File: mod/friends/classes/Elgg/Friends/Menus/Filter.php [hunk 3]
```diff
@@ -1,24 +1,51 @@
...
+	
+		$vars = $hook->getParam('vars');
+		$selected = $hook->getParam('selected');
+		$type = $hook->getType();
+	
+		$items = $hook->getValue();
+		$items[] = \ElggMenuItem::factory([
+			'name' => 'friend',
+			'text' => elgg_echo('friends'),
+			'href' => (isset($vars['friend_link'])) ? $vars['friend_link'] : "$type/friends/{$user->username}",
```
Comment (jeabakker) [inline mod/friends/classes/Elgg/Friends/Menus/Filter.php]: shouldn't we add a check for friends route? No idea if enough information is present

File: mod/groups/classes/Elgg/Groups/Group.php [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+class Group {
+
+	/**
+	 * Populates the ->getUrl() method for group objects
+	 *
+	 * @param \Elgg\Hook $hook 'entity:url', 'group'
+	 *
+	 * @return void|string
+	 */
+	public static function getEntityUrl(\Elgg\Hook $hook) {
```
Comment (jeabakker) [inline mod/groups/classes/Elgg/Groups/Group.php]: why? this is handled in the `view:group:group` route

File: mod/groups/classes/Elgg/Groups/Menus/Site.php [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+	 * @param \Elgg\Hook $hook 'register', 'menu:site'
+	 */
+	public static function register(\Elgg\Hook $hook) {
+		$return = $hook->getValue();
+		
+		$return[] = \ElggMenuItem::factory([
+			'name' => 'groups',
+			'icon' => 'users',
+			'text' => elgg_echo('groups'),
+			'href' => elgg_generate_url('collection:group:group:all'),
```
Comment (jeabakker) [inline mod/groups/classes/Elgg/Groups/Menus/Site.php]: `default:group:group` route

File: mod/groups/classes/Elgg/Groups/PageOwner.php [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+class PageOwner {
+
+	/**
+	 * Helper handler to correctly resolve page owners on group routes
+	 *
+	 * @param \Elgg\Hook $hook "page_owner", "system"
+	 *
+	 * @return int|void
+	 */
+	public static function detectPageOwner(\Elgg\Hook $hook) {
```
Comment (jeabakker) [inline mod/groups/classes/Elgg/Groups/PageOwner.php]: this was reported in a different ticket to verify if still needed?

File: mod/groups/lib/functions.php [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+ * Get the access collection for a given group
+ *
+ * @param \ElggGroup $group the group
+ *
+ * @return \ElggAccessCollection|false
+ *
+ * @internal
+ * @since 3.0
+ */
+function _groups_get_group_acl(\ElggGroup $group) {
```
Comment (jeabakker) [inline mod/groups/lib/functions.php]: Why a helper function?

File: mod/likes/lib/functions.php [hunk 8]
```diff
@@ -0,0 +1,87 @@
+<?php
+use Elgg\Likes\DataService;
```
Comment (jeabakker) [inline mod/likes/lib/functions.php]: .....

File: mod/pages/classes/Elgg/Pages/Menus/PagesNav.php [hunk 9]
```diff
@@ -1,22 +1,23 @@
 <?php
 
-namespace Elgg\Pages;
-
-use Elgg\Menu\MenuItems;
```
Comment (jeabakker) [inline mod/pages/classes/Elgg/Pages/Menus/PagesNav.php]: this was probably used for the return value of a function

File: mod/search/classes/Elgg/Search/Site.php [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+	public static function preventSearchIndexing(\Elgg\Hook $hook) {
+		$rules = [
+			'',
+			'User-agent: *',
+			'Disallow: /search/',
+			''
+		];
+	
+		$text = $hook->getValue();
+		$text .= implode("\r\n", $rules);
```
Comment (jeabakker) [inline mod/search/classes/Elgg/Search/Site.php]: why not plain string, and implode with `PHP_EOL`

File: mod/system_log/classes/Elgg/SystemLog/Logger.php [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+	/**
+	 * Disables the logging
+	 *
+	 * @param \Elgg\Event $event 'all', 'all'
+	 *
+	 * @return true
+	 */
+	public static function disableLogging(\Elgg\Event $event) {
+		// disable the system log for upgrades to avoid exceptions when the schema changes.
+		elgg_unregister_event_handler('log', 'systemlog', 'system_log_default_logger');
```
Comment (jeabakker) [inline mod/system_log/classes/Elgg/SystemLog/Logger.php]: is this the correct callback

File: mod/system_log/classes/Elgg/SystemLog/Logger.php [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+		elgg_unregister_event_handler('all', 'all', 'system_log_listener');
```
Comment (jeabakker) [inline mod/system_log/classes/Elgg/SystemLog/Logger.php]: same

File: mod/thewire/classes/Elgg/TheWire/Menus/Site.php [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+	 * @return void|\Elgg\Collections\Collection
+	 */
+	public static function register(\Elgg\Hook $hook) {
+		
+		$return = $hook->getValue();
+		$return[] = \ElggMenuItem::factory([
+			'name' => 'thewire',
+			'icon' => 'comments-o',
+			'text' => elgg_echo('thewire'),
+			'href' => elgg_generate_url('collection:object:thewire:all'),
```
Comment (jeabakker) [inline mod/thewire/classes/Elgg/TheWire/Menus/Site.php]: `default:object:thewire` see earlier remarks

File: mod/uservalidationbyemail/classes/Elgg/UserValidationByEmail/Menus/UserUnvalidatedBulk.php [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+			return;
+		}
+		
+		$return = $hook->getValue();
+		
+		$return[] = \ElggMenuItem::factory([
+			'id' => 'uservalidationbyemail-bulk-resend',
+			'name' => 'uservalidationbyemail:resend:bulk',
+			'text' => elgg_echo('uservalidationbyemail:admin:resend_validation'),
+			'href' => 'action/uservalidationbyemail/resend_validation',
```
Comment (jeabakker) [inline mod/uservalidationbyemail/classes/Elgg/UserValidationByEmail/Menus/UserUnvalidatedBulk.php]: `elgg_generate_action_url()`

**Review by jeabakker — CHANGES_REQUESTED** [review]

File: mod/groups/classes/Elgg/Groups/PageOwner.php (hunk 6 above)
Comment (jdalsem) [reply mod/groups/classes/Elgg/Groups/PageOwner.php]: yep... #13064

File: mod/groups/lib/functions.php (hunk 7 above)
Comment (jdalsem) [reply mod/groups/lib/functions.php]: i do not know... you can create a ticket to investigate

File: mod/search/classes/Elgg/Search/Site.php (hunk 10 above)
Comment (jdalsem) [reply mod/search/classes/Elgg/Search/Site.php]: do not want to change this as part of this PR

File: mod/system_log/classes/Elgg/SystemLog/Logger.php (hunk 11 above)
Comment (jdalsem) [reply mod/system_log/classes/Elgg/SystemLog/Logger.php]: good catch

File: mod/system_log/classes/Elgg/SystemLog/Logger.php (hunk 12 above)
Comment (jdalsem) [reply mod/system_log/classes/Elgg/SystemLog/Logger.php]: same

File: mod/groups/lib/functions.php (hunk 7 above)
Comment (jeabakker) [reply mod/groups/lib/functions.php]: #13091

File: mod/groups/classes/Elgg/Groups/Group.php (hunk 4 above)
Comment (jdalsem) [reply mod/groups/classes/Elgg/Groups/Group.php]: https://github.com/Elgg/Elgg/issues/13092

File: mod/friends/classes/Elgg/Friends/Menus/Filter.php (hunk 3 above)
Comment (jdalsem) [reply mod/friends/classes/Elgg/Friends/Menus/Filter.php]: no idea

Comment (jeabakker) [reply mod/friends/classes/Elgg/Friends/Menus/Filter.php]: #13093

File: docs/tutorials/widget.rst [hunk 15]
```diff
@@ -40,24 +40,20 @@ Registering your widget
...
-    }
-
-    return function() {
-        elgg_register_event_handler('init', 'system', 'hello_init');
-    }
+    	return [
+		    'widgets' => [
+				'helloworld' => [
+					'name' => 'Hello, world!',
+					'description' => 'The "Hello, world!" widget',
```
Comment (jeabakker) [inline docs/tutorials/widget.rst]: should we maybe explain the 'magic' widget title/description languages keys?

**Review by jeabakker — APPROVED** [review]

**Pull request merged**